core/src/ledger/parameters/storage.rs

```diff
@@ -20,10 +21,19 @@ pub fn is_parameter_key(key: &Key) -> bool {
 
 /// Returns if the key is a protocol parameter key.
 pub fn is_protocol_parameter_key(key: &Key) -> bool {
+    // TODO: improve this code; use some kind of prefix
+    // tree to efficiently match `key`
     is_epoch_duration_storage_key(key)
         || is_max_expected_time_per_block_key(key)
         || is_tx_whitelist_key(key)
         || is_vp_whitelist_key(key)
+        || is_implicit_vp_key(key)
+        || is_epochs_per_year_key(key)
+        || is_pos_gain_p_key(key)
+        || is_pos_gain_d_key(key)
+        || is_staked_ratio_key(key)
+        || is_pos_inflation_amount_key(key)
+        || is_max_proposal_bytes_key(key)
```

Comment:
oic the efficiency issue now. It would be good I guess if all protocol parameters were under some subtree e.g. `/parameters/protocol/...` or something, then we could easily recognize them

Reply:
I was playing around with trie codegen over the weekend for this specific use-case, but I still need to improve the generated code. https://gist.github.com/sug0/1bc632eb19ebdeb2d596c1617fba4138

Reply:
btw, we do have the same internal address as a prefix in each of these keys